packages/react-core/src/components/Modal/ModalBox.tsx

```diff
@@ -36,10 +42,17 @@ export const ModalBox: React.FunctionComponent<ModalBoxProps> = ({
     className={css(
       styles.modalBox,
       className,
+      position === 'top' && styles.modifiers.alignTop,
       variant === 'large' && styles.modifiers.lg,
       variant === 'small' && styles.modifiers.sm,
       variant === 'medium' && styles.modifiers.md
     )}
+    {...(distance && {
+      style: {
+        '--pf-c-modal-box--m-align-top--spacer': distance,
+        ...props.style
```

Comment:
I think you should destructure `style` instead of relying on the `{...(distance && {}}` tag being after `{...props}`.

Then you could write
```jsx
style={{
  ...(distance && { '--pf-c-modal-box--m-align-top--spacer': distance }),
  ...style
}}
```